src/trade-params/saga/DurationSaga.js

```diff
@@ -142,7 +142,9 @@ export function* handleStartTimeChange(action) {
     const contractPerType = contractNeeded[tradeCategory][type];
     const durationAllowed = allTimeRelatedFieldValid(newDateStart, duration, durationUnit, contractPerType);
 
-    if (durationAllowed) {
+     if (isValidTime(time)) {
+        yield put(updateTradeError(index, 'durationError', 'Time format is wrong!'));
```

Comment:
Note that yield is not return, if you want to exit early, explicit `return` is needed

@nuruddeensalihu